src/OrleansConsulUtils/ConsulBasedMembershipTable.cs

```diff
@@ -0,0 +1,202 @@
+        /// Consul Membership Provider does not support the extended Membership Protocol,
+        /// therefore there is no MembershipTable to Initialise
+        /// </remarks>
+        public Task InitializeMembershipTable(GlobalConfiguration config, Boolean tryInitTableVersion, TraceLogger traceLogger)
+        {
+            Init(config.DeploymentId, config.DataConnectionString, traceLogger);
+
+            return TaskDone.Done;
+        }
+
+        private void Init(String deploymentId, String dataConnectionString, TraceLogger traceLogger)
+        {
+            _logger = traceLogger;
+            _deploymentId = deploymentId;
+            _connectionString = dataConnectionString;
+
+            _consulClient =
+                new ConsulClient(
+                    new ConsulClientConfiguration
+                    {
+                        Address = new Uri(dataConnectionString)
+                    });
+        }
+
+        public async Task<MembershipTableData> ReadRow(SiloAddress siloAddress)
+        {
+            var siloRegistration = await GetConsulSiloRegistration(siloAddress);
+
+            return AssembleMembershipTableData(siloRegistration);
+        }
+
+        public async Task<MembershipTableData> ReadAll()
+        {
+            var deploymentKVAddresses = await _consulClient.KV.List(ConsulSiloRegistration.ParseDeploymentKVPrefix(_deploymentId));
+            if (deploymentKVAddresses.Response == null)
+            {
+                _logger.Verbose("Could not find any silo registrations for deployment {0}.", _deploymentId);
+                return new MembershipTableData(null);
+            }
+
+            var allSiloRegistrations = deploymentKVAddresses.Response.Select(siloKV => ConsulSiloRegistration.FromKVPair(_deploymentId, siloKV)).ToArray();
+
+            return AssembleMembershipTableData(allSiloRegistrations);
+        }
+
+        public async Task<Boolean> InsertRow(MembershipEntry entry, TableVersion tableVersion)
+        {
+            try
+            {
+                //Ensure the KV for the entry does not already exist
+                var value = await _consulClient.KV.Get(ConsulSiloRegistration.ParseDeploymentSiloKey(_deploymentId, entry.SiloAddress));
+                if (value.Response != null)
+                {
+                    _logger.Verbose("ConsulMembershipProvider could not insert a registration for silo {0}, because one already exists.", entry.SiloAddress);
+                    return false;
+                }
+
+                var consulSiloKVEntry = ConsulSiloRegistration.FromMembershipEntry(_deploymentId, entry, tableVersion.VersionEtag);
+                await _consulClient.KV.Put(consulSiloKVEntry.ToKVPair());
+
+                return true;
+            }
+            catch (Exception ex)
+            {
+                _logger.Verbose("ConsulMembershipProvider failed to insert registration for silo {0}; {1}.", entry.SiloAddress, ex);
+                return false;
```

Comment:
You should not return false when the table write has failed for non semantically reasons.
replace `return false;` with `throw`